### ISSUE TITLE
MaxAsync does not work on IdentityUserClaim<string> and IdentityRoleClaim<string> records

### ISSUE THREAD

**Issue by adambarath** [MoonriseSoftwareCalifornia/AspNetCore.Identity.CosmosDb#14]
**Describe the bug**
MaxAsync does not work on IdentityUserClaim<string> and IdentityRoleClaim<string> records.

**To Reproduce**
Following code produces SQL like
  -  https://github.com/MoonriseSoftwareCalifornia/AspNetCore.Identity.CosmosDb/blob/491f5460c6e7bee60bca5964813bbdc07a3c6211/AspNetCore.Identity.CosmosDb/Stores/CosmosRoleStore.cs#L316
  - https://github.com/MoonriseSoftwareCalifornia/AspNetCore.Identity.CosmosDb/blob/491f5460c6e7bee60bca5964813bbdc07a3c6211/AspNetCore.Identity.CosmosDb/Stores/CosmosUserStore.cs#L801 

`SELECT MAX(c["Id"]) AS m FROM root c WHERE(c["Discriminator"] = "IdentityUserClaim<string>")`
'Id' field is stored as string value, therefore the MAX operation does not work as expected.
Example data set: "Id": "9", "Id": "10"

**Current behavior** max result is "9" instead of "10".

**Expected behavior**
**Current behavior** max result has to be "10" instead of "9".

**Screenshots**
![image](https://github.com/MoonriseSoftwareCalifornia/AspNetCore.Identity.CosmosDb/assets/14049972/15c00211-dafe-409b-bc6c-d7fa5f89f3ec)
![image](https://github.com/MoonriseSoftwareCalifornia/AspNetCore.Identity.CosmosDb/assets/14049972/979e2d04-9388-4de1-866d-bd4f7aacb2f0)

**Desktop (please complete the following information):**
- TargetFramework net6.0
 - Version="2.1.4"

**Additional context**
I tried to add claims to a user record and discovered this issue.

**Comment by toiyabe** [comment]
Hello @adambarath, I just applied a patch that hopefully solves the problem.  Now I first pull the keys out of Cosmos and in doing so convert to an int array.  Now, outside of Cosmos, the Max function should work as the values are integers.  If you have time, please have a look? Thanks for your input!

**Comment by adambarath** [comment]
Hi [toiyabe](https://github.com/toiyabe) , 

There is another thing I found in Net8.0 today.

So if someone will upgrade from for eg. net6 to net8 is going to have an issue. There will be no claims read from cosmosdb. The reason of this issue is the value of Discriminators is different in Net8 EF Core, than was in Net6 EF Core.

So the example:
Net6.0 ``` "Discriminator": "IdentityUserClaim<string>"```
Net8.0 ``` "Discriminator": "IdentityUserClaim"```

EF is generates the following where clause of course:  ```where "Discriminator" = "IdentityUserClaim"```. 

If you starts a project and seed database from Net8.0 it is not a problem at all. Just the upgraded projects gonna have this issue.

**Comment by toiyabe** [comment]
@adambarath, any suggestion(s) on how to make this backward compatible?

**Comment by toiyabe** [comment]
@adambarath, I tried my suggestion above, did not work. Failed to run and pass tests. Any thoughts?

**Comment by adambarath** [comment]
[toiyabe](https://github.com/toiyabe), in my project I migrated the whole database via scipts. Export, fix data, batch import...

I do not know this area of EF Core. 

Aweful idea, but what about a loud and clear disclaimer in the readme? 🥇

**Comment by toiyabe** [comment]
Hello @adambarath,

I think I found a solution here as I believe I can set the discriminator value as shown below.  This does not break the tests that I have so far, and I pushed up the code change. Perhaps let me know if this is back-ward compatible now?

Role claim discriminator value can be set like this:
`
builder
    .HasDiscriminator().HasValue("IdentityRoleClaim<string>"); // Backward compatibility.
`

and user claim like this,

`
builder
                .HasDiscriminator().HasValue("IdentityUserClaim<string>"); // Backward compatibility.
`
